main.py

```diff
@@ -89,6 +90,9 @@ async def on_message(message: Message):
     if message.author == client.user:
         return
 
+    if not isinstance(message.author, Member):
+        return
```

Comment:
`message.author` being a `User` is a side effect of the message appearing outside of a guild (i.e. `message.author.roles` is needed to check the user is authorised in the guild. 

Currently we don't allow any commands over DM, which is really the reason we're returning early here.

Essentially, we might want to be more explicit and check whether `message.guild` is `None`, rather than have the check revolve around a side effect.